### PR TITLE
Global options and arcomplete support

### PR DESCRIPTION
**Pull request by MarioSchwalbe**

### CODE REVIEW

File: mando/core.py [hunk 1]
```diff
@@ -30,6 +30,25 @@ def __init__(self, prog=None, version=None, **kwargs):
...
+            "Positional arguments not supported here"
+        completer = kwd.pop('completer', None)
+        arg = self.parser.add_argument(*args, **kwd)
+        if completer is not None:
+            arg.completer = completer
+        return arg
+
+    # Attribute lookup fallback redirecting to (internal) options instance.
+    def __getattr__(self, attr):
+        return getattr(self.options, attr)
```
Comment (rubik) [inline mando/core.py]: When is this used in the code?

File: mando/core.py [hunk 2]
```diff
@@ -119,7 +142,16 @@ def parse(self, args):
         then be called as ``command(*args)``.
 
         :param args: The arguments to parse.'''
-        arg_map = self.parser.parse_args(args).__dict__
+        try:
+            # run completion handler before parsing
+            import argcomplete
+            argcomplete.autocomplete(self.parser)
+        except ImportError:  # pragma: no cover
```
Comment (rubik) [inline mando/core.py]: I looked into `argcomplete` and it looks good. Keeping it as an optional dependency seems like the correct choice here. I'm opening a new issue to remember to mention it in the documentation.

File: mando/core.py (hunk 1 above)
Comment (MarioSchwalbe) [inline mando/core.py]: That's a convenience accessor to get the global options. Supports something like that:
```
Script = mando.core.Program()
Script.option("--foo", dest='bar')
```
And in some of the subcommands (after actual argument parsing):
```
print(Script.bar)
```
Note the different var name (if not deduced automatically by ArgumentParser()).
Besides that it's not used internally, and hence, might not be included in the tests yet.
Please also note the comment in my last e-mail about attribute names.

**Comment by rubik** [comment]
Perfect, it looks good and I can merge it. Thanks for your contribution!

**Comment by MarioSchwalbe** [comment]
Cool. You're welcome.

**Pull request merged**